adapters/eplanning/eplanning.go

```diff
@@ -117,24 +118,61 @@ func (adapter *EPlanningAdapter) MakeRequests(request *openrtb.BidRequest, reqIn
 		pageDomain = defaultPageURL
 	}
 
-	uri := adapter.URI + fmt.Sprintf("/%s/%s/%s/%s?r=pbs&ncb=1&ur=%s&e=%s", clientID, dfpClientID, pageDomain, sec, url.QueryEscape(pageURL), strings.Join(spacesStrings, "+"))
+	var requestTarget string
+	if request.App != nil && request.App.Bundle != "" {
+		requestTarget = request.App.Bundle
+	} else {
+		requestTarget = pageDomain
+	}
+
+	uriObj, err := url.Parse(adapter.URI)
+	if err != nil {
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	uriObj.Path = uriObj.Path + fmt.Sprintf("/%s/%s/%s/%s", clientID, dfpClientID, requestTarget, sec)
+	query := url.Values{}
+	query.Set("r", "pbs")
+	query.Set("ncb", "1")
+	if request.App == nil {
+		query.Set("ur", url.QueryEscape(pageURL))
```

Comment:
You don't need the `url.QueryEscape` call here since you're now using the `query` object. `query.Encode()` will escape the values. Consider adding a test case for when the pageURL contains a character which needs to be escaped.